### PR TITLE
 Adding interface

### PR DESCRIPTION
**Pull request by karnadattasai**
Adding Interface. Just an outline.

### CODE REVIEW

File: cache.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package main
+
+type Cache interface {
+	read()
```
Comment (priyanshukm) [inline cache.go]: For an interface to be exposed the functions inside it should be exported. All these are private as of now. Please make them public/exported

File: cache.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package main
```
Comment (priyanshukm) [inline cache.go]: Also you should create this interface in a different package. Say `service/cache`

File: cache.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+type Cache interface {
+	read()
+	write()
+	new()
+}
+type CacheBase struct {
```
Comment (priyanshukm) [inline cache.go]: I am not sure what Base will be used here for ? Can you describe more about it's use ?

File: cache.go [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package main
+
+type Cache interface {
+	read()
+	write()
+	new()
```
Comment (priyanshukm) [inline cache.go]: Can you please add comments about these functions and it's usage. Also are you sure that we don't require any arguments for it ? As per my understanding `write()` must be passed some data to writer in the cache right ?

File: cache.go (hunk 1 above)
Comment (karnadattasai) [reply cache.go]: Yeah will do that.

**Review by priyanshukm — CHANGES_REQUESTED** [review]
Please resolve these comments

File: cache.go (hunk 2 above)
Comment (karnadattasai) [reply cache.go]: Ok, will do that.

**Comment by priyanshukm** [comment]
Also please have commit messages like `[learn-go-cache] Added interface`. Let's have this prefix `[learn-go-cache]` for all the commits you are creating. Also I am not sure, how you are doing the changes after review, but just a suggestion that, `ammend` the commit on which you have comments instead of creating a new one. 
It's something like `git commit --amend` once you have done `git add .` for all the local changes.

**Review by priyanshukm — APPROVED** [review]
Looks neat

**Pull request merged**